src/sph-layout.cc

```diff
@@ -0,0 +1,225 @@
+
+        std::cout.width(12);
+        std::cout.precision(3);
+        std::cout << std::fixed << box.getLatMin();
+        std::cout << " | ";
+
+        std::cout.width(12);
+        std::cout.precision(3);
+        std::cout << std::fixed << box.getLatMax();
+        std::cout << " | ";
+
+        std::cout.width(12);
+        std::cout.precision(6);
+        std::cout << std::fixed << box.area();
+        std::cout << " | ";
+
+        Chunk2WorkerMap::const_iterator const & itr = chunk2worker.find(chunkId);
+        const std::string worker{itr == chunk2worker.end() ? "" : itr->second};
+
+        std::cout.width(10);
+        std::cout << worker;
+
+        std::cout << "\n";
+    }
+
+    // The general description of the tool reported by the argument parser
+    // when runing the tool with option "--help".
+    char const * help =
+        "The tool will report a layout of partitions for the specified\n"
+        "configuration of stripes and overlaps.\n";
+}
+
+int main(int argc, char const * const * argv) {
+
+    try {
+        po::options_description desc("\\_______________ Layout", 80);
+        desc.add_options()
+
+        ("help,h",    ::help)
+        ("verbose,v", "Produce verbose output.")
+
+        ("part.num-stripes",     po::value<int>()   ->default_value(85),    "Chunk file name prefix.")
+        ("part.num-sub-stripes", po::value<int>()   ->default_value(12),    "The number of sub-stripes to divide each stripe into.")
+        ("part.overlap",         po::value<double>()->default_value(0.01),  "Chunk/sub-chunk overlap radius (deg).")
+
+        ("chunk2worker", po::value<std::string>(), "Chunk-to-worker map.")
+
+        ("chunk", po::value<std::vector<int32_t>>(), "Chunk identifier.");
+
+        po::positional_options_description pos_descr;
+        pos_descr.add("chunk", -1);
+
+        po::variables_map vm;
+        po::store(
+            po::command_line_parser(argc, argv).options(desc).positional(pos_descr).run(),
+            vm);
+        po::notify(vm);
+
+        const bool verbose = vm.count("verbose") > 0;
+        
+        const int    numStripes             = vm["part.num-stripes"]    .as<int>();
+        const int    numSubStripesPerStripe = vm["part.num-sub-stripes"].as<int>();
+        const double overlap                = vm["part.overlap"]        .as<double>();
+
+        if (verbose) {
+            std::cout
+                << "\n"
+                << "** Configuration **\n"
+                << "\n"
+                << "  part.num-stripes:     " << numStripes << "\n"
+                << "  part.num-sub-stripes: " << numSubStripesPerStripe << "\n"
+                << "  part.overlap:         " << overlap << "\n"
+                << std::endl;
+        }
+
+        std::vector<int32_t> chunks =
+            vm.count("chunk") ? vm["chunk"].as<std::vector<int32_t>>() : std::vector<int32_t>();
+
+        if (chunks.empty())
+            for (int32_t chunkId = 0 ; chunkId < 20000; ++chunkId)
```

Comment:
Re-implemented by adding two command line parameters to specify a range of chunk IDs in case if no specific IDs were passed to the application.